pennylane/transforms/mitigate.py

```diff
@@ -215,13 +319,27 @@ def circuit(w1, w2):
             [apply(m) for m in tape.measurements]
         out_tapes.append(t)
 
+    # print(qml.drawer.tape_text(out_tapes[1].expand(), decimals=3))
+
     def processing_fn(results):
-        """Maps from input tape executions to an error-mitigated estimate"""
-        results = [
-            results[i : i + reps_per_factor] for i in range(0, len(results), reps_per_factor)
-        ]  # creates nested list according to reps_per_factor
-        results = mean(results, axis=1)
-        extrapolated = extrapolate(scale_factors, results, **extrapolate_kwargs)
+        # print(results)
+        # """Maps from input tape executions to an error-mitigated estimate"""
+        # results = [
+        #     results[i : i + reps_per_factor] for i in range(0, len(results), reps_per_factor)
+        # ]  # creates nested list according to reps_per_factor
+        # print(results)
+        # results_flattened = mean(results, axis=1)
+        # print(results_flattened)
+
+        results_flattened = []
+        for i in range(0, len(results), reps_per_factor):
+           results_flattened.append(mean(qml.math.stack(results[i : i + reps_per_factor]), axis=0))
```

Comment:
This part is a bit tricky because you can have two different kinds of possible nestings: One from `reps_per_factor > 1` and one from having multiple outputs in the qnode. I wonder if this can be solved more elegantly.